ext/include/opentelemetry/ext/http/client/http_client.h

```diff
@@ -110,13 +110,127 @@ struct cmp_ic
+                 nostd::string_view input_ssl_client_key_path,
+                 nostd::string_view input_ssl_client_key_string,
+                 nostd::string_view input_ssl_client_cert_path,
+                 nostd::string_view input_ssl_client_cert_string,
+                 nostd::string_view input_ssl_min_tls,
+                 nostd::string_view input_ssl_max_tls,
+                 nostd::string_view input_ssl_cipher,
+                 nostd::string_view input_ssl_cipher_suite)
+      : use_ssl(false),
+        ssl_insecure_skip_verify(input_ssl_insecure_skip_verify),
+        ssl_ca_cert_path(input_ssl_ca_cert_path),
+        ssl_ca_cert_string(input_ssl_ca_cert_string),
+        ssl_client_key_path(input_ssl_client_key_path),
+        ssl_client_key_string(input_ssl_client_key_string),
+        ssl_client_cert_path(input_ssl_client_cert_path),
+        ssl_client_cert_string(input_ssl_client_cert_string),
+        ssl_min_tls(input_ssl_min_tls),
+        ssl_max_tls(input_ssl_max_tls),
+        ssl_cipher(input_ssl_cipher),
+        ssl_cipher_suite(input_ssl_cipher_suite)
+  {
+    /* Use SSL if url starts with "https:" */
+    if (strncmp(url.data(), "https:", 6) == 0)
+    {
+      use_ssl = true;
+    }
+  }
+
+  /**
+    Use HTTPS (true) or HTTP (false).
+  */
+  bool use_ssl{false};
+  /**
+    Skip SSL/TLS verifications.
+    Setting this flag to true is insecure.
+  */
+  bool ssl_insecure_skip_verify{false};
+  /**
+    Path to the CA CERT file.
+  */
+  std::string ssl_ca_cert_path{};
+  /**
+    CA CERT.
+    Used only if @p ssl_ca_cert_path is empty.
+  */
+  std::string ssl_ca_cert_string{};
+  /**
+    Path to the client key file.
+  */
+  std::string ssl_client_key_path{};
+  /**
+    Client key.
+    Used only if @p ssl_client_key_path is empty.
+  */
+  std::string ssl_client_key_string{};
+  /**
+    Path to the client cert file.
+  */
+  std::string ssl_client_cert_path{};
+  /**
+    Client cert.
+    Used only if @p ssl_client_cert_path is empty.
+  */
+  std::string ssl_client_cert_string{};
+
+  /**
+    Minimum SSL version to use.
+    Valid values are:
+    - empty (no minimum version required)
+    - "1.0" (TLSv1.0)
+    - "1.1" (TLSv1.1)
+    - "1.2" (TLSv1.2)
+    - "1.3" (TLSv1.3)
+  */
+  std::string ssl_min_tls{};
+
+  /**
+    Maximum SSL version to use.
+    Valid values are:
+    - empty (no maximum version required)
+    - "1.0" (TLSv1.0)
+    - "1.1" (TLSv1.1)
+    - "1.2" (TLSv1.2)
+    - "1.3" (TLSv1.3)
+  */
+  std::string ssl_max_tls{};
+
+  /**
+    TLS Cipher.
+    This is for TLS 1.0, 1.1 and 1.2.
+    The list is delimited by colons (":").
+    Cipher names depends on the underlying CURL implementation.
+  */
+  std::string ssl_cipher{};
+
+  /**
+    TLS Cipher suite.
+    This is for TLS 1.3.
+    The list is delimited by colons (":").
+    Cipher names depends on the underlying CURL implementation.
+  */
+  std::string ssl_cipher_suite{};
```

Comment:
Thanks for the comments.

Working on a fix, but delayed by other items, hoping to address this soon.

Reply:
@lalitb Thanks for the review.

For cases when SSL may or may not be used (OTLP HTTP), I don't expect the runtime overhead to be an issue.

For cases when SSL is never used with HTTP, in particular for exporters that do not support SSL (like Zipkin), I agree there should be a way to opt out of SSL entirely, if someone does not use the OTLP HTTP exporter.

For TLS, it is also desirable to have a way to out out.

As a result, implemented the following changes:

* Flag ENABLE_OTLP_HTTP_SSL_PREVIEW is still used, but the scope is only for the OTLP HTTP exporter
* Flag ENABLE_OTLP_HTTP_SSL_TLS_PREVIEW is still used, but the scope is only for the OTLP HTTP exporter
* Flag ENABLE_HTTP_SSL_PREVIEW is added, the scope is the SSL support in the HTTP client
* Flag ENABLE_HTTP_SSL_TLS_PREVIEW is added, the scope is the SSL TLS support in the HTTP client 

This makes it possible to use ENABLE_HTTP_SSL_PREVIEW for another http client, should it become necessary, independently of the otlp http exporter, and likewise for ENABLE_HTTP_SSL_TLS_PREVIEW.

The code has a lot if ifdef/endif because of this, which is tedious, but the overall structure is regular and should be still maintainable.

Tested the following build locally, in maintainer mode:

* SSL + TLS
* SSL only
* no SSL